### PR TITLE
Add prometheus scrape information to probe list, so that envoy does n…

### PR DESCRIPTION
**Pull request by objectiser**
…ot trace the scrape requests.

Follow on from #6068, to include prometheus scrape requests in the list of probes that Envoy should not trace.

### CODE REVIEW

File: pilot/pkg/networking/plugin/health/health.go [hunk 1]
```diff
@@ -61,7 +61,9 @@ func buildHealthCheckFilters(filterChain *plugin.FilterChain, probes model.Probe
 		// Check that the probe matches the listener port. If not, then the probe will be handled
 		// as a management port and not traced. If the port does match, then we need to add a
 		// health check filter for the probe path, to ensure that health checks are not traced.
-		if probe.Port.Port == endpoint.Port {
+		// If no probe port is defined, then port has not specifically been defined, so assume filter
+		// needs to be applied.
+		if probe.Port == nil || probe.Port.Port == endpoint.Port {
```
Comment (rshriram) [inline pilot/pkg/networking/plugin/health/health.go]: If port is nil how will kube pick the port ?

File: pilot/pkg/serviceregistry/kube/controller.go [hunk 2]
```diff
@@ -353,6 +364,30 @@ func (c *Controller) WorkloadHealthCheckInfo(addr string) model.ProbeList {
...
+				}
+			}
+		}
+		if pod.Annotations[PrometheusPath] != "" {
+			path = pod.Annotations[PrometheusPath]
+		}
+		probes = append(probes, &model.Probe{
+			Port: port,
+			Path: path,
+		})
```
Comment (rshriram) [inline pilot/pkg/serviceregistry/kube/controller.go]: This is smart. To make Prometheus APIs as health checks

File: pilot/pkg/networking/plugin/health/health.go (hunk 1 above)
Comment (objectiser) [reply pilot/pkg/networking/plugin/health/health.go]: As I understand it, if the container has multiple ports then prometheus will try to scrape from each of them unless the `prometheus.io/port` annotation is defined. So to cover this situation, this logic would result in a health check filter being created for each port.

**Comment by objectiser** [comment]
@rshriram Had to rebase due to conflict on istio/deps. Could you reapply lgtm and add ok-to-test. Thanks.

**Pull request merged**